packages/protocol/contracts/L1/verifiers/SGXVerifier.sol

```diff
@@ -7,38 +7,165 @@
 pragma solidity ^0.8.20;
 
 import { EssentialContract } from "../../common/EssentialContract.sol";
+import { ECDSAUpgradeable } from "@ozu/utils/cryptography/ECDSAUpgradeable.sol";
 import { Proxied } from "../../common/Proxied.sol";
 
 import { TaikoData } from "../TaikoData.sol";
 
 import { IVerifier } from "./IVerifier.sol";
 
 /// @title GuardianVerifier
-// TODO(dani): implement this verifier.
 contract SGXVerifier is EssentialContract, IVerifier {
-    uint256[50] private __gap;
+    event InstanceAdded(
+        uint256 indexed instanceId, address indexed instanceInitPubKey
+    );
 
-    error UNIMPLEMENTED();
+    struct ProofData {
+        uint256 id;
+        address newPubKey;
```

Comment:
[Here](https://mbed-tls.readthedocs.io/en/latest/security-advisories/mbedtls-security-advisory-2020-04/) is a description.
To avoid leakage, we need to change it relatively frequently.
<img width="707" alt="kép" src="https://github.com/taikoxyz/taiko-mono/assets/51912515/da666d07-4361-4a81-8537-339170e7a0dd">

Reply:
> I'm not familiar with SGX, SGX can automatically generate a new pubkey after each run? 
it should, yes!